test/functional/lua/fs_spec.lua

```diff
@@ -529,4 +526,53 @@ describe('vim.fs', function()
       end)
     end
   end)
+
+  describe('relpath()', function()
+    it('works', function()
+      local cwd = assert(t.fix_slashes(assert(vim.uv.cwd())))
+
+      eq(nil, vim.fs.relpath('/var/lib', '/var'))
+      eq(nil, vim.fs.relpath('/var/lib', '/bin'))
+      eq(nil, vim.fs.relpath('/var/lib', 'bin'))
+      eq(nil, vim.fs.relpath('/var/lib', './bin'))
+      eq(nil, vim.fs.relpath('/var/lib', '././'))
+      eq(nil, vim.fs.relpath('/var/lib', '../'))
+      eq(nil, vim.fs.relpath('/var/lib', '/'))
+      eq(nil, vim.fs.relpath('/var/lib', '//'))
+      eq(nil, vim.fs.relpath(' ', '/var'))
+      eq(nil, vim.fs.relpath(' ', '/var'))
+      eq('.', vim.fs.relpath('/var/lib', '/var/lib'))
+      eq('lib', vim.fs.relpath('/var/', '/var/lib'))
+      eq('var/lib', vim.fs.relpath('/', '/var/lib'))
+      eq('bar/package.json', vim.fs.relpath('/foo/test', '/foo/test/bar/package.json'))
+      eq('foo/bar', vim.fs.relpath(cwd, 'foo/bar'))
+      eq('foo/bar', vim.fs.relpath('.', vim.fs.joinpath(cwd, 'foo/bar')))
+      eq('bar', vim.fs.relpath('foo', 'foo/bar'))
+      eq(nil, vim.fs.relpath('/var/lib', '/var/library/foo'))
+
+      if is_os('win') then
+        eq(nil, vim.fs.relpath('/', ' '))
+        eq(nil, vim.fs.relpath('/', 'var'))
+      else
+        local cwd_rel_root = cwd:sub(2)
+        eq(cwd_rel_root .. '/ ', vim.fs.relpath('/', ' '))
+        eq(cwd_rel_root .. '/var', vim.fs.relpath('/', 'var'))
+      end
+
+      if is_os('win') then
+        eq(nil, vim.fs.relpath('c:/aaaa/', '/aaaa/cccc'))
+        eq(nil, vim.fs.relpath('c:/aaaa/', './aaaa/cccc'))
+        eq(nil, vim.fs.relpath('c:/aaaa/', 'aaaa/cccc'))
+        eq(nil, vim.fs.relpath('c:/blah\\blah', 'd:/games'))
+        eq(nil, vim.fs.relpath('c:/aaaa/bbbb', 'c:/aaaa'))
+        eq('cccc', vim.fs.relpath('c:/aaaa/', 'c:/aaaa/cccc'))
+        eq('aaaa/bbbb', vim.fs.relpath('c:/', 'c:\\aaaa\\bbbb'))
```

Comment:
use mixed casing to drive home the point that drive-letter casing is ignored
```suggestion
        eq('aaaa/bbbb', vim.fs.relpath('C:/', 'c:\\aaaa\\bbbb'))
```